### PR TITLE
Implement AWS S3 fetcher

### PR DESCRIPTION
**Pull request by ari-aviran**
**What does this PR do?**
Implements an AWS S3 fetcher for CIS AWS benchmark.

**Related Issues**
https://github.com/elastic/cloudbeat/issues/540

Related security policies PR - https://github.com/elastic/csp-security-policies/pull/156

**Checklist**
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

**Review by uri-weisman — CHANGES_REQUESTED** [review]
Nice job! 💪 
I've added few comments...

File: .gitignore [hunk 1]
```diff
@@ -13,6 +13,7 @@ cloudbeat-test
 *local.yaml
 *local.yml
 fields.yml
+.hermit/
```
Comment (oren-zohar) [inline .gitignore:16]: :100:

File: deploy/aws/cloudbeat-aws.yml [hunk 2]
```diff
@@ -2,10 +2,12 @@ cloudbeat:
   type: cloudbeat/cis_aws
   access_key_id: ${AWS_ACCESS_KEY_ID:""}
   secret_access_key: ${AWS_SECRET_ACCESS_KEY:""}
+  default_region: eu-west-1
```
Comment (oren-zohar) [inline deploy/aws/cloudbeat-aws.yml:5]: this field is not part of our integration [configuration](https://github.com/elastic/integrations/blob/main/packages/cloud_security_posture/data_stream/findings/agent/stream/aws.yml.hbs), no?

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: magefile.go [hunk 3]
```diff
@@ -346,5 +346,7 @@ func BuildOpaBundle() error {
 		return err
 	}
 
+	cspPoliciesPkgDir = "/Users/ari.aviran/dev/elastic/csp-security-policies"
```
[the remark this answers is not in the document]
Comment (ari-aviran) [reply magefile.go]: Removed, I'll update the tag of CSP security policies before merging

File: resources/providers/awslib/s3_provider.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+func NewS3Provider(cfg aws.Config, log *logp.Logger) *S3Provider {
+	client := s3.NewFromConfig(cfg)
+	return &S3Provider{
+		log,
+		client,
+	}
+}
+
+func (p S3Provider) DescribeS3Buckets(ctx context.Context) ([]AwsResource, error) {
+	clientBuckets, err := p.client.ListBuckets(ctx, &s3.ListBucketsInput{})
```
[the remark this answers is not in the document]
Comment (ari-aviran) [reply resources/providers/awslib/s3_provider.go:53]: @uri-weisman where did you find this limit? As far as I can tell there is no limit to this API call ([there is one for `ListObjects`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html))

File: deploy/cloud/README.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # Cloud Deployment
 
-**Motivation**  
+**Motivation**
```
[the remark this answers is not in the document]
Comment (ari-aviran) [reply deploy/cloud/README.md]: Reverted

File: deploy/aws/cloudbeat-aws.yml (hunk 2 above)
Comment (ari-aviran) [reply deploy/aws/cloudbeat-aws.yml:8]: Yes it should not be part of the UI, this is useful for setting a specific region for testing until we'll have [multi-region support](https://github.com/elastic/cloudbeat/issues/592). Anyway reverted for now :)

File: resources/providers/awslib/s3_provider.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (uri-weisman) [reply resources/providers/awslib/s3_provider.go:53]: @ari-aviran you're right.
I probably got confused with `ListObjects`

File: beater/cloudbeat.go [hunk 6]
```diff
@@ -104,7 +104,8 @@ func newCloudbeat(_ *beat.Beat, cfg *agentconfig.C) (*cloudbeat, error) {
 		return nil, err
 	}
 
-	data, err := fetchersManager.NewData(log, c.Period, time.Minute, fetchersRegistry)
+	// TODO: timeout should be configurable and not hard-coded. Setting to 5 minutes for now to account for CSPM fetchers
```
[the remark this answers is not in the document]
Comment (ari-aviran) [reply beater/cloudbeat.go:107]: https://github.com/elastic/cloudbeat/issues/653

File: resources/providers/awslib/s3/provider_test.go [hunk 7]
```diff
@@ -0,0 +1,152 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
```
Comment (uri-weisman) [inline resources/providers/awslib/s3/provider_test.go:1]: Nice tests 🚀

**Review by uri-weisman — CHANGES_REQUESTED** [review]

File: resources/providers/awslib/s3/provider.go [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+	if err != nil {
+		log.Errorf("Could not list s3 buckets: %v", err)
+		return nil, err
+	}
+
+	var result []awslib.AwsResource
+
+	for _, clientBucket := range clientBuckets.Buckets {
+		bucketRegion := p.getBucketRegion(ctx, clientBucket.Name)
+		if bucketRegion == p.region {
```
[the remark this answers is not in the document]
Comment (ari-aviran) [reply resources/providers/awslib/s3/provider.go:53]: Added comments to this flow

**Review by uri-weisman — APPROVED** [review]

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**